template.go

```diff
@@ -57,7 +57,15 @@ func (r {{$typename}}) getString() (string, error) {
     if s, ok := interface{}(r).(fmt.Stringer); ok {
         return s.String(), nil
     }
+    {{if $.Stringer}}
     return r.ToString()
+    {{else}}
+    s, ok := _{{$typename}}ValueToName[r]
```

Comment:
Why are we duplicating this code between here and `ToString` instead of extracting it to a common private function they can both use?